components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/UserMappingDAOImplIT.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.core.dao.impl;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.carbon.apimgt.core.dao.UserMappingDAO;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+
+public class UserMappingDAOImplIT extends DAOIntegrationTestBase {
+
+    @Test(description = "Get all tags")
+    public void testUserNameMapping() throws APIManagementException {
+        UserMappingDAO userMappingDAO = DAOFactory.getUserMappingDAO();
+        String pseudoName = userMappingDAO.getPseudoNameByUserID("test_user");
+        Assert.assertNotNull(pseudoName);
+        Assert.assertTrue(userMappingDAO.getUserIDByPseudoName(pseudoName).equalsIgnoreCase("test_user"));
```

Comment:
This integration test is sure to fail since we have not saved a user named "test_user" in the DB

Reply:
If user name mapping was not there then we will add it and return. Hence this will pass. It was tested with H2, oracle etc and works fine